index.js

```diff
@@ -157,6 +152,92 @@ app.get('/logout', (req, res) => {
 });
 
+app.get('/dashboard', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+
+    try {
+        let allLocations = await db.getLocations()
+        
+        let viewParams = {
+            locations: allLocations
+                .map(location => { 
+                    return { name: location.displayName, id: location.locationid }
+                })
+        }
+        viewParams.viewMessage = allLocations.length >= 1 ? 'Please select a location' : 'No locations to display'
+
+        res.send(Mustache.render(locationsDashboardTemplate, viewParams))
+    }
+    catch(err) {
+        console.log(err)
+        res.status(500).send()
+    }
+})
+
+app.get('/dashboard/:locationId', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+    else if(typeof req.params.locationId !== "string") {
+        let locations = await db.getlocations()
+        res.redirect('/dashboard/' + locations[0].id)
+        return
+    }
+
+    try {
+        let recentSessions = await db.getHistoryOfSessions(req.params.locationId)
+        let currentLocation = await db.getLocationData(req.params.locationId)
+        let allLocations = await db.getLocations()
+        
+        let viewParams = {
+            recentSessions: [],
+            currentLocationName: currentLocation.display_name,
+            locations: allLocations
+                .map(location => { 
+                    return { name: location.displayName, id: location.locationid }
+                })
+        }
+
+        // commented out keys were not shown on the old frontend but have been included in case that changes
+        for(const recentSession of recentSessions) {
+            let startTime = moment(recentSession.startTime, moment.ISO_8601)
+                .tz('America/Vancouver')
+                .format('DD MMM Y, hh:mm:ss A')
+            let endTime = recentSession.endTime 
+                ? moment(recentSession.endTime, moment.ISO_8601)
+                    .tz('America/Vancouver')
+                    .format('DD MMM Y, hh:mm:ss A')
+                : 'Ongoing'
+
+            viewParams.recentSessions.push({
+                // locationid: recentSession.locationid, // this one was deemed redundant, remove entirely?
+                startTime: startTime,
+                endTime: endTime,
+                // odFlag: recentSession.odFlag,
+                state: recentSession.state,
+                // phonenumber: recentSession.phonenumber,
+                notes: recentSession.notes,
+                incidentType: recentSession.incidentType,
+                sessionid: recentSession.sessionid,
+                duration: recentSession.duration,
+                // stillCounter: recentSession.stillCounter,
+                chatbotState: recentSession.chatbotState,
+                // alertReason: recentSession.alertReason,
```

Comment:
This would actually be a nice one to have on the dashboard, I think.

The others that you have commented out, I'd just delete them.

Reply:
which one? I see 2 in the little snippet here (still counter and alert reason)

Reply:
`alertReason`

Reply:
Okay cool thanks, I will update that 👍

Reply:
Ah, I see that you removed the other comments, which is good. But I also would suggest that we add `alertReason` to the front-end.